### PR TITLE
Add .dockerignore

### PR DESCRIPTION
**Pull request by eunchan**
Copied .gitignore to .dockerignore to reduce the copy size when
triggering the docker run.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+opentitan-docs
+
+# fusesoc result
+build/
+
+# build directories with special suffixes
+build*/
+
+# Sphinx result
+_build/
```
Comment (imphil) [inline .dockerignore]: Where is that coming from?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+*~
+*.bak
+__pycache__
+opentitan-docs
+
+# fusesoc result
+build/
+
+# build directories with special suffixes
+build*/
```
Comment (gkelly) [inline .dockerignore]: This catches `build/` doesn't it?

Comment (eunchan) [reply .dockerignore]: Oh.. yes. I will delete above Fusesoc ignore list :)

File: .dockerignore (hunk 1 above)
Comment (eunchan) [reply .dockerignore]: ... Maybe the buildthedocs trial? I will remove this line :)

**Pull request merged**